go/pkg/bertyprotocol/keystore_device.go

```diff
@@ -243,6 +248,40 @@ func (a *deviceKeystore) RestoreAccountKeys(sk crypto.PrivKey, proofSK crypto.Pr
 	return nil
 }
 
+func (a *deviceKeystore) AttachmentPrivKey(cid ipfscid.Cid) (crypto.PrivKey, error) {
+	return a.ks.Get(cid.String())
+}
+
+func (a *deviceKeystore) PutAttachmentPrivKey(cid ipfscid.Cid, sk crypto.PrivKey) error {
+	return a.ks.Put(cid.String(), sk)
+}
+
+func (a *deviceKeystore) PutAttachments(attachments []*bertytypes.Attachment) error {
+	for _, at := range attachments {
+		cid, err := ipfscid.Cast(at.GetCID())
+		if err != nil {
+			return errcode.ErrDeserialization.Wrap(err)
+		}
+
+		s := at.GetSecret()
+		if len(s) <= len(attachmentKeyV0Prefix) || !bytes.Equal(s[:len(attachmentKeyV0Prefix)], []byte(attachmentKeyV0Prefix)) {
+			return errcode.ErrDeserialization.Wrap(fmt.Errorf("invalid secret prefix %s", s))
+		}
```

Comment:
I think this kind of check should be done in `api_attachment.go` or `attachment_utils.go` to centralize the logic of choosing the right alogrithm when multikey will be ready.
IMO this `PutAttachments` should just take two arrays of crypto.PrivKey and cIDs.

Reply:
agreed